bugsnag/configuration.py

```diff
@@ -104,6 +95,295 @@ def __init__(self):
 
         self.runtime_versions = {"python": platform.python_version()}
 
+    def configure(self, **options):
+        """
+        Validate and set configuration options. Will warn if an option is of an
+        incorrect type.
+        """
+        # Overrides the default implementation to provide warnings for unknown
+        # options. In the __future__ this will not be necessary by instead
+        # making configure() enumerate all of the options as kwargs rather than
+        # allowing arbitrary input.
+        configurable_options = [
+            'api_key', 'app_version', 'asynchronous', 'auto_notify',
+            'auto_capture_sessions', 'delivery', 'endpoint', 'hostname',
+            'ignore_classes', 'lib_root', 'notify_release_stages',
+            'params_filters', 'project_root', 'proxy_host', 'release_stage',
+            'send_code', 'session_endpoint', 'traceback_exclude_modules',
+            'use_ssl',
+        ]
+
+        for option_name in options.keys():
+            if option_name not in configurable_options:
+                message = 'received unknown configuration option "{0}"'
+                warnings.warn(message.format(option_name), RuntimeWarning)
```

Comment:
This is maybe a BC break — if someone was storing arbitrary data on the Configuration instance then this change would break that, so it might be safer to always do the `setattr` call

I'm happy to make this change as I think it's far more likely for this to be done in error (e.g. a typo in a config value name)